### DISCUSSION TITLE
GPU_VIEWER 2.00 - Release Notes

### DISCUSSION THREAD

**Discussion by arunsivaramanneo** [arunsivaramanneo/GPU-Viewer#54] [Show and tell]
## Thank You All

Hi All,

Before i  start mentioning the change log below, I want to wish you all a **Merry Chrismas and a Happy New Year**. Its been a really active year for this app. I really had to put lot of time in getting the Security Improvement  #49 done . Also, when doing this i decided to move the Application to Gtk 4 (note 4.0 not 4.1 or 4.x) (Gnome developers are still trying to figure out what they want, as they keep depreciating the features). Anyway i still went ahead and ported it to Gtk 4.0 and in  future i will definitely move it to the latest Gtk 4.X as and when i get time. So to all , try out the application and do let me know what you think and ofcourse  if there are any issues, as i could only test with the devices which i have (though covered all major Vendor Intel, Nvidia and AMD.)

***Thanks for trying out the application and If you find the project interesting enough, please consider making a donation. Even a small one would mean the world to me.***

[![Donate](https://user-images.githubusercontent.com/30646692/209199473-a51dbd6c-d7f2-4bfe-a392-0abf20cbc4ec.png)](https://www.paypal.com/donate/?hosted_button_id=7M3PMM78FBR4Q)

Thank You
Arun Sivaraman

## Change Log

1. Application now uses Gtk4.0, those who want to use the Gtk3.0 version, please use the version 1.43 Zip. So basically Ubuntu 22.04 or above
3. Security improvements (i guess) - Moved from Python OS to Python Subprocess
4. Tweaked gtk.css file for minor UI enhancement, users who are clonning are recommended to install latest version of ***Orchis-gtk-theme*** or **Materia-gtk-theme** . Through PPA and AUR, Orchis-gtk-theme will be installed automatically
5. Added Vulkan Instance version to be displayed in Vulkan Tab
6.  Queues Tab in Vulkan now correctly displays the Video Encode and Decode support (Note : Latest Nvidia vulkan developer drivers are needed or upcoming mesa update with Vulkan Video support)
7.  Ton of Code cleanup (Did my best to cleanup the mess , i guess this will be ongoing process)
8.  Tested with Vulkan Version 1.3.238, Mesa utils- 8.4+, Clinfo v3.0.21.02.21, vdpauinfo(supports X11 only) v1.5-1
9.  Tested with Nvidia Geforce GTX 1050 Ti , AMD Radeon Vega 8 and Intel HD Graphics. 
10. Tested on Ubuntu 22.10+ , Manjaro 22.00

## Known Issues:
1. The ComboboxText doesn't seem to be working on Ubuntu 22.04 (Gnome 42.5), but works fine on Ubuntu 22.10 or above (Gnome 43 or above). I still trying to figure out what the issue as i don't why its not working properly below Gnome 43, if you guys have any luck please do let me know
2. Dark Mode is still a no no , so please install Orchis or Materia or even adwaita theme to show the details correctly

## GPU-Viewer v2.0

![Screenshot from 2022-12-23 09-53-49](https://user-images.githubusercontent.com/30646692/209314443-dc2cb6f2-98cd-4cc2-98c1-26bc82d320f3.png)

*The experience may vary as per the theme you are using.

<hr /><em>This discussion was created from the release <a href='https://github.com/arunsivaramanneo/GPU-Viewer/releases/tag/v2.02'>GPU_VIEWER 2.00 - Release Notes </a>.</em>

**Comment by Danny3** [comment]
Thank you very much for this release!

A couple of questions:

What does  Gtk 4 mean of people who use KDE Plasma, how is the compatibility with it?
I'm curios if the system's window decorations are still respected.
If the dark theme preference is respected.
If the fonts look as crisp as in the programs built with Gtk 3 or Qt.
Is it possible to publish this in Debian's repository so that Debian 12 is released we can simply install it from there?

**Reply by arunsivaramanneo** [reply]
Well, This App right now works with Light themes, ( damn dark mode :( ! i will get it done one day).  I have tested  this application on Manjaro 22 KDE  and it works fine... I will post the screenshot someday today.  I am not forcing Fonts to the users as they will have there own preference, so whatever font you have configured on your system, the same font will be used for displaying the content (in the screenshot I am using fonts-Roboto) 

I will try to publish that in Debian's repository (heard its a long process), but not sure when this can be done. I will add it to the list. Also, the application is available in Flathub to download and install https://flathub.org/apps/details/io.github.arunsivaramanneo.GPUViewer

**Reply by Danny3** [reply]
Not having a dark mode at the moment it's ok, don't worry about it!
If it worked ok on Manjaro 22 DE then it should work ok on other distros with KDE too.
No worries about screenshots, do them only if you have time and want to. For me your word is enough.
As for font's I was asking about the fact that I heard that Gtk 4 might not be so good with fonts as they removed some some sub-pixel alignment, rendering or how it's called an might not be as crisp or sharp as in Gtk 3 or Qt.
Maybe I'm mistaken about this.

Here on KDE Plasma I think this works as I just changed the sub-pixel rendering from None to RGB and I think that the fonts just became more sharp, especially the small ones which are easier to read.

As for publishing it into the Debian's repository being a long process, I think you're right as I recently asked another developer about its very cool tool about the same thing and he told me this:

https://github.com/JonMagon/KDiskMark/issues/110

So with a long queue / time to wait it's important also when you make the request.

If the request it's done now, maybe it can be fulfilled in 2-3 years.
I know that's pretty bad, but I think it's better than nothing and once in Debian it can reach many distros without problems.

But in the meantime it's very good that we have it on Flathub too.

Thank you very much!

**Reply by arunsivaramanneo** [reply]
Thank you , here is the one i tested in Manjaro KDE. I am trying to work on my CSS as i have little to no knowledge on how to do it, if i am able to get that done . i think i should be able to get the dark mode / render the app on other Distro's just fine. Thanks a lot for your feeback/ideas .. I will see what i can do 

![Screenshot_20221223_120943](https://user-images.githubusercontent.com/30646692/209334830-74cb3a22-a78c-4be8-a333-9a8b2eef7447.png)

**Comment by jeffriechers** [comment]
So doing some vgpu testing for pass through in Citrix Sesssions.  Running this on Rocky 9.1 and all the font characters are boxes instead of active letters.  Is there a special font that is needed or coded for in this app?

**Comment by arunsivaramanneo** [comment]
Hi, i have not hard coded any font , whatever font user applies system wide
should be used ... can you post any screenshots. Also, how you have
installed the application i mean through PPA, Aur , Flathub or cloned ?

**Comment by jeffriechers** [comment]
Installed from Flatpack.  
![image](https://user-images.githubusercontent.com/71444494/212736814-edfc2876-a821-4409-97c4-24fce5778c25.png)

![image](https://user-images.githubusercontent.com/71444494/212736870-a3aeb03e-d4c7-4555-8ce8-932454deb0c2.png)

**Comment by arunsivaramanneo** [comment]
Can you try clonning and installing and see if you see the same issue? ...
personally i used Roboto-Font in ubuntu but i have not made it mandatory to
install it...

**Comment by tinywrkb** [comment]
@jeffriechers what fonts are loaded by the application? For example, this is what's loaded on my system.

```
$ flatpak run --env=FC_DEBUG=1 io.github.arunsivaramanneo.GPUViewer  | grep file

        file: "/usr/share/fonts/liberation-fonts/LiberationSans-Regular.ttf"(s)
        file: "/usr/share/fonts/liberation-fonts/LiberationSans-Regular.ttf"(w)
        file: "/usr/share/fonts/liberation-fonts/LiberationSans-Bold.ttf"(s)
        file: "/usr/share/fonts/liberation-fonts/LiberationSans-Bold.ttf"(w)
        file: "/usr/share/fonts/liberation-fonts/LiberationSans-Bold.ttf"(s)
        file: "/usr/share/fonts/liberation-fonts/LiberationSans-Bold.ttf"(s)
        file: "/usr/share/fonts/liberation-fonts/LiberationSans-Bold.ttf"(w)
        file: "/usr/share/fonts/liberation-fonts/LiberationSans-Regular.ttf"(s)
        file: "/usr/share/fonts/liberation-fonts/LiberationSans-Bold.ttf"(w)
        file: "/usr/share/fonts/liberation-fonts/LiberationSans-Regular.ttf"(w)
```

Did you create any override to load your own fontconfig config files?

**Comment by jeffriechers** [comment]
I'm seeing this.

	file: "/usr/share/fonts/cantarell/Cantarell-Bold.otf"(s)

(python3:2): Pango-WARNING **: 13:07:50.979: failed to create cairo scaled font, expect ugly output. the offending font is 'Cantarell Bold 11'

(python3:2): Pango-WARNING **: 13:07:50.979: font_face status is: file not found

(python3:2): Pango-WARNING **: 13:07:50.979: scaled_font status is: file not found

(python3:2): Pango-WARNING **: 13:07:50.980: failed to create cairo scaled font, expect ugly output. the offending font is 'Cantarell Bold 11'

(python3:2): Pango-WARNING **: 13:07:50.980: font_face status is: file not found

(python3:2): Pango-WARNING **: 13:07:50.980: scaled_font status is: file not found
	file: "/usr/share/fonts/cantarell/Cantarell-Bold.otf"(w)

(python3:2): Pango-WARNING **: 13:07:50.980: failed to create cairo scaled font, expect ugly output. the offending font is 'Cantarell 11'

(python3:2): Pango-WARNING **: 13:07:50.980: font_face status is: file not found

**Comment by tinywrkb** [comment]
Please post the output of the following commands when running directly on the system, meaning not in a Flatpak sandbox.

```
$ gsettings get org.gnome.desktop.interface font-name
$ gsettings get org.gnome.desktop.interface document-font-name
$ gsettings get org.gnome.desktop.interface monospace-font-name
```

**Comment by jeffriechers** [comment]
'Cantarell 11'
'Cantarell 11'
'Monospace 11'

From the looks of it this is pretty common with flatpak.  Seeing lots of google posts about it, but none of the remediation is fixing it.  Still hunting.

**Reply by tinywrkb** [reply]
> 'Cantarell 11' 'Cantarell 11' 'Monospace 11'


I believe these are the defaults for GTK, so it should not trigger such issue as the one you're hitting.  
On my system, if I change the GTK font to Cantarell, then the variable variant `Cantarell-VF.otf` is used.  
Actually, there's no ` Cantarell-Bold.otf` font in the runtime, so I wonder if fontconfig is wrong or if a different runtime is mounted. 

What's the output of
```
$ flatpak info $(flatpak info --show-runtime io.github.arunsivaramanneo.GPUViewer)
```

And when running the app
```
$ flatpak ps --columns=application,runtime,runtime-branch,runtime-commit | grep GPUViewer
```

And last
```
$ flatpak run --command=file io.github.arunsivaramanneo.GPUViewer /usr/share/fonts/cantarell/Cantarell-Bold.otf
```

**Comment by jeffriechers** [comment]
Ok, I was able to fix it.  What I did was install Gnome-Tweak and changed my Interface and Document text to Carlito Regular.  Very odd, but seems to be something Flatpak particular.  It's specific with that Cantarell font.  It doesn't like scaling, which Flatpak seems to have built in.
